test/Npgsql.Tests/Types/EnumTests.cs

```diff
@@ -384,6 +387,26 @@ await using (var __ = await GetTempTableName(conn, out var table))
             }
         }
 
+        [Test]
+        public async Task WriteUnmappedEnum()
+        {
+            await using var conn = await OpenConnectionAsync();
+            await using var _ = DeferAsync(() => conn.ExecuteNonQueryAsync("DROP TYPE IF EXISTS mood_unmapped"));
+            await conn.ExecuteNonQueryAsync($"CREATE TYPE mood_unmapped AS ENUM ('sad', 'ok', 'happy')");
+
+            conn.ReloadTypes();
+
+            await using var cmd = new NpgsqlCommand($"SELECT @p", conn)
+            {
+                Parameters = { new("p", MoodUnmapped.Happy) }
+            };
+
+            await using var reader = await cmd.ExecuteReaderAsync();
```

Comment:
Shouldn't we convert the enum parameter to text in PostgreSQL, and assert on that? You yourself made the very correct point that doing read and write in the same test is not good :)